src/Collector.cs

```diff
@@ -153,5 +143,26 @@ private static Socket Connect()
       _log.Info("Connected to collector.");
       return socket;
     }
+
+    private static bool IsSocketDisconnected (Socket socket)
+    {
+      // Is there any data available?
+      byte[] buffer = null;
+      while (socket.Poll(1, SelectMode.SelectRead))
+        {
+          // If no data is available then socket disconnected
+          if (socket.Available == 0)
+            return true;
```

Comment:
socket.Poll with SelectRead will report true only if the socket is currently being listened to (highly unlikely, as it is owned by this worker thread!) or if there is data available, or if the socket is closed.  I do not know what set of circumstances led Undead to do it this way, but basically this is saying

```
while(socket has data available to read OR is closed/disconnected)
  if(the amount of data available is zero)
   conclude the socket must therefore be closed
```

This whole method is ripe for a refactoring, for this and other reasons which I'm sure I'm about to discuss, look for it in #11